src/transaction_builder.js

```diff
@@ -153,33 +172,252 @@ export class TransactionBuilder {
    * @see TimeoutInfinite
    */
   setTimeout(timeout) {
+    if (timeout < 0) {
+      throw new Error('timeout cannot be negative');
+    }
+
+    return this.setMaxTime(Math.floor(Date.now() / 1000) + timeout);
+  }
+
+  /**
+   * If you want to prepare a transaction which will become valid at some point in the future, you can set a minimum time precondition.
+   * Internally this will set the <code>minTime</code> precondition.
+   *
+   * @param {Date|number} dateOrEpochSeconds Either a js Date object, or a number of unix epoch seconds. Can't be negative.
+   * If the value is `0`, the transaction is valid immediately.
+   * @return {TransactionBuilder}
+   * @see TransactionPreconditions
+   */
+  setMinTime(dateOrEpochSeconds) {
```

Comment:
Wait let's keep it simple and non-breaking: don't allow setting min or max time individually. Let's keep `setTimeout` and `setTimebounds` as before.

Reply:
`setTimebounds`?

Reply:
ah, conflating my SDKs... just `setTimeout`, then